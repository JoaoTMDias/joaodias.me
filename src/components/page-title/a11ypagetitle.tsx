// Libraries
import * as React from "react";
import styled from "styled-components";

// Component Props
interface IA11yPageTitleProps {
	title: string;
}

/**
 * @description Accessible Page Title for Reach Router
 * @author  Jo√£o Dias
 * @date  29/November/2018 at 23:42
 * @extends {React.FunctionComponent}
 */
export const A11yPageTitle: React.FunctionComponent<IA11yPageTitleProps> = props => {
	const { title } = props;
	return (
		<Title
			id="a11y-page-title"
			data-testid="a11y-page-title"
			role="heading"
			aria-level={1}
			aria-live="polite"
			aria-atomic="true"
			tabIndex={-1}
		>
			{title}
		</Title>
	);
};

const Title = styled.h1`
	clip: rect(1px, 1px, 1px, 1px);
	height: 1px;
	overflow: hidden;
	position: absolute !important;
	width: 1px;
	opacity: 0;
`;

A11yPageTitle.defaultProps = {
	title: "Page Title",
};

export default A11yPageTitle;
