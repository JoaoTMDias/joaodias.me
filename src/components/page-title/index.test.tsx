import React from "react";
import { render, cleanup } from "@testing-library/react";
import { PageTitle } from "./index";
import { A11yPageTitle } from "./a11ypagetitle";

afterEach(cleanup);

describe("<PageTitle>", () => {
	const expected = {
		title: "title",
		subtitle: "subtitle",
		center: true,
		isProject: true,
		backgroundColor: "var(--color-white, #ffffff)",
	};

	it("should render without error", () => {
		const component = render(<PageTitle backgroundColor={expected.backgroundColor} />);
		expect(component).toMatchSnapshot();
	});

	it("should render a title and a subtitle", async () => {
		const { findByTestId } = render(
			<PageTitle title={expected.title} subtitle={expected.subtitle} backgroundColor={expected.backgroundColor} />,
		);
		const title = await findByTestId("title");
		const subtitle = await findByTestId("subtitle");

		expect(title).toHaveTextContent(expected.title);
		expect(subtitle).toMatchSnapshot(expected.subtitle);
	});

	it("should render class if isProject", async () => {
		const { findByTestId } = render(
			<PageTitle isProject={expected.isProject} backgroundColor={expected.backgroundColor} />,
		);
		const row = await findByTestId("row");
		expect(row).toHaveClass("isProject");
	});

	it("should center the page title", async () => {
		const { findByTestId } = render(<PageTitle center={expected.center} backgroundColor={expected.backgroundColor} />);
		const row = await findByTestId("row");

		expect(row).toHaveClass("isCenter");
	});
});

describe("<A11yPageTitle />", () => {
	const expected = {
		title: "Accessible Title",
	};

	it("should render without error", async () => {
		const component = render(<A11yPageTitle title={expected.title} />);
		const title = await component.findByTestId("a11y-page-title");
		expect(title).toHaveTextContent(`${expected.title}`);
	});
});
