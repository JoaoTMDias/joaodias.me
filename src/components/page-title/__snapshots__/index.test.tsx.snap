// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageTitle> should render a title and a subtitle: subtitle 1`] = `
<h2
  class="sc-AxiKw eaRcaI"
  data-testid="subtitle"
  style="opacity: 0; transform: translate3d(0, 1rem, 0);"
  tabindex="0"
>
  subtitle
</h2>
`;

exports[`<PageTitle> should render without error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="sc-AxjAm kHCYAy layout__container layout__container layout__section utilities--above-the-fold"
      >
        <div
          class="sc-AxirZ gVtcha layout__row parallax"
          data-testid="row"
        >
          <h1
            class="sc-AxhCb gvVZKr"
            data-testid="title"
            style="opacity: 0; transform: translate3d(0, 1rem, 0);"
          />
          <h2
            class="sc-AxiKw eaRcaI"
            data-testid="subtitle"
            style="opacity: 0; transform: translate3d(0, 1rem, 0);"
            tabindex="0"
          />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sc-AxjAm kHCYAy layout__container layout__container layout__section utilities--above-the-fold"
    >
      <div
        class="sc-AxirZ gVtcha layout__row parallax"
        data-testid="row"
      >
        <h1
          class="sc-AxhCb gvVZKr"
          data-testid="title"
          style="opacity: 0; transform: translate3d(0, 1rem, 0);"
        />
        <h2
          class="sc-AxiKw eaRcaI"
          data-testid="subtitle"
          style="opacity: 0; transform: translate3d(0, 1rem, 0);"
          tabindex="0"
        />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
